framework/web/View.php

```diff
@@ -470,6 +456,57 @@ public function registerJs($js, $position = self::POS_READY, $key = null)
         }
     }
 
+    /**
+     * Registers a JS or CSS file.
+     *
+     *
+     * @param string $url the JS file to be registered.
+     * @param string $type type (js or css) of the file.
+     * @param array $options the HTML attributes for the script tag. The following options are specially handled
+     * and are not treated as HTML attributes:
+     * @param string $key the key that identifies the JS script file. If null, it will use
+     * $url as the key. If two JS files are registered with the same key at the same position, the latter
+     * will overwrite the former. Note that position option takes precedence, thus files registered with the same key,
+     * but different position option will not override each other.
+     * @throws InvalidConfigException
+     */
+    private function registerFile($type, $url, $options = [], $key = null)
+    {
+        $url = Yii::getAlias($url);
```

Comment:
```suggestion

```

Reply:
In this case, the tests will not be passed. The $url can contain another alias FA `@web/assetSources/js/missing-file.js`

Reply:
I've meant moving it: https://github.com/yiisoft/yii2/pull/16826#discussion_r322979844

Reply:
Yes, but what about if we want to register $url`@web/assetSources/js/missing-file.js`?
thus the file path will be `@webroot/@web/assetSources/js/missing-file.js` and wont be found...